kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt

```diff
@@ -108,15 +108,7 @@ internal fun Type.asTypeName(
               nullableProjection?.let { projection ->
                 when (projection) {
                   Type.Argument.Projection.IN -> WildcardTypeName.supertypeOf(argumentTypeName)
-                  Type.Argument.Projection.OUT -> {
-                    if (argumentTypeName == ANY) {
-                      // This becomes a *, which we actually don't want here.
-                      // List<Any> works with List<*>, but List<*> doesn't work with List<Any>
```

Comment:
I've removed this because it is wrong in general. The logic works for `List` only because it has declaration-site `out` variance. And `out Any` should never be printed as `*`, see https://github.com/square/kotlinpoet/issues/520